mycroft/skills/core.py

```diff
@@ -199,6 +202,38 @@ def __init__(self, name, emitter=None):
         self.reload_skill = True
         self.events = []
 
+    @staticmethod
+    def on_intent_failure(message):
+        """Goes through all fallback handler until one returns true"""
+        for _, handler in sorted(MycroftSkill.fallback_handlers.items(),
+                                 key=operator.itemgetter(0)):
+            if handler(message):
+                return
+        logger.warn('No fallback could handle intent.')
+
+    @staticmethod
+    def register_fallback(handler, priority):
+        """
+        Register a function to be called as a general info fallback
+        Fallback should receive message and return
+        a boolean (True if succeeded or False if failed)
+
+        Lower priority gets run first
+        0 for high priority 100 for low priority
+        """
+        while priority in MycroftSkill.fallback_handlers:
+            priority += 1
+
+        MycroftSkill.fallback_handlers[priority] = handler
```

Comment:
Well, it should keep incrementing the priority in the two lines of code above until there is no handler at that priority.